### PR TITLE
Add table format as parameter

### PR DESCRIPTION
**Pull request by sbs-jms**
Addresses a number of issues in the creation of SwordfishTypes.md, and allows table formatting to be a parameter. Closes DMTF/Redfish-Tools#427

### CODE REVIEW

**Comment by sbs-jms** [comment]
Closes SNIA/SSM#1938 and SNIA/SSM#1873

**Comment by mraineri** [comment]
@mraineri to look at this offline to see how this affects existing docs DMTF produces

**Comment by sbs-jms** [comment]
@mraineri, I think it was decided, in one of the joint calls with @rahlvers, that RF wasn't interested in this change is that right, or is there pending change somewhere? If so, will it be in place for SSM 1.2.5 / RF 2023.1, or should I rely on a forked clone of the tools branch for now?

**Comment by mraineri** [comment]
@sbs-jms no, that's not the case; we didn't fully understand what the change does, and I was supposed to follow up with you to talk about it, but it fell off my radar

**Comment by sbs-jms** [comment]
Ok, thanks for refreshing my memory. Do you have some time in the next week or so to walk through the proposed changes?

**Comment by mraineri** [comment]
Sure, I'll send you an email so we can chat about it 1 on 1

**Comment by mraineri** [comment]
@mraineri will compare the output of the docs before and after the change to see how Redfish docs are affected; the Travis check indicates there is something different, but from looking over the code it might just be some extra new lines, which are harmless

**Comment by mraineri** [comment]
@sbs-jms there are two changes I see in the Redfish docs that I wasn't expecting (picture of the PDF output attached).

The first one is an extra blank line where the red circle is. Is this intentional? I think it's introduced where I added "See comment" inline with the diff.

The second change redoes the output of the Action URI line. While it's different than what we publish, I actually like this change. The only thing I think we should change to it is make "Base URI of target resource" italicized. Thoughts on this?

@jautor, what do you think?

doc

File: doc-generator/doc_formatter/markdown_generator.py [hunk 1]
```diff
@@ -530,7 +549,7 @@ def format_action_parameters(self, schema_ref, prop_name, prop_descr, action_par
 
         if deprecated_descr:
             formatted.append(self.formatter.para(italic(deprecated_descr)))
-        formatted.append(self.formatter.para(self.formatter.bold(_("Description"))))
+        formatted.append(self.formatter.para('&nbsp;\n\n' + self.formatter.bold(_("Description"))))
```
Comment (mraineri) [inline doc-generator/doc_formatter/markdown_generator.py:552]: See main comment

**Comment by jautor** [comment]
@mraineri @sbs-jms I agree that the linebreak for Action URI is a good improvement, and yes, please snip that extra blank line.  And we should drop the colon at the end of that "Action URI:" heading now for consistency - it was there as a separator.

**Comment by mraineri** [comment]
Done; just need to follow-up on whether the extra blank space is intentional or not.

**Comment by sbs-jms** [comment]
I don't think the extra blank line is required either; good catch. It should be removed.

**Pull request merged**